generators/csharp/codegen/src/context/generation-info.ts

```diff
@@ -0,0 +1,876 @@
+
+export type TRelativeFilePath = string & {
+    __RelativeFilePath: void;
+};
+export type TAbsoluteFilePath = string & {
+    __AbsoluteFilePath: void;
+};
+/**
+ * Central configuration and code generation context for C# SDK generation.
+ *
+ * This class serves as the single source of truth for all code generation decisions,
+ * consolidating names, namespaces, types, and settings into a unified graph. It eliminates
+ * duplication between dynamic and regular generators by providing consistent access to
+ * generation metadata.
+ *
+ * ## Key Responsibilities:
+ * - Manages namespace organization for generated C# code
+ * - Provides standardized naming conventions for classes, types, and identifiers
+ * - Lazily initializes and caches type references and configuration values
+ * - Serves as a navigation hub for the IR (Intermediate Representation) model
+ * - Coordinates between code generation utilities (CSharp, NameRegistry, etc.)
+ *
+ * ## Lazy Initialization:
+ * All returned values use lazy initialization for performance:
+ * - Parameterless functions: Evaluated once and cached on first access
+ * - Parameterized functions: Remain as functions, evaluated on each call
+ *
+ * ## Usage Pattern:
+ * ```typescript
+ * const generation = new Generation(ir, apiName, config, generatorConfig);
+ * const namespace = generation.namespaces.root; // Lazy-evaluated and cached
+ * const pager = generation.types.Pager(itemType); // Function, evaluated each call
+ * ```
+ *
+ * This object is safe to pass throughout the code generation pipeline and should be
+ * the primary source for all naming and type reference decisions.
+ */
+export class Generation {
+    /**
+     * Creates a new Generation context for C# code generation.
+     *
+     * @param intermediateRepresentation - The Fern IR containing the API definition,
+     *                                     supports both static and dynamic IR formats
+     * @param apiName - The name of the API being generated (used for namespace/class naming)
+     * @param customConfig - User-provided custom configuration overrides for the generator
+     * @param generatorConfig - Core generator configuration including organization and workspace info
+     * @param logger - expose logging functions to generation classes
+     */
+    constructor(
+        private readonly intermediateRepresentation:
+            | IntermediateRepresentation
+            | FernIr.dynamic.DynamicIntermediateRepresentation,
+        private readonly apiName: string,
+        private readonly customConfig: BaseCsharpCustomConfigSchema,
+        private readonly generatorConfig: FernGeneratorExec.config.GeneratorConfig,
+        private readonly makeRelativeFilePath: (path: string) => TRelativeFilePath = (path) =>
+            path as TRelativeFilePath,
+        private readonly makeAbsoluteFilePath: (path: string) => TAbsoluteFilePath = (path) =>
+            path as TAbsoluteFilePath,
+        public readonly logger: {
```

Comment:
There should be a common type for this somewhere that you can reuse.
It looks a little wild to have this type inline here.